code/go/0chain.net/smartcontract/storagesc/allocation.go

```diff
@@ -379,8 +395,39 @@ func (sc *StorageSmartContract) newAllocationRequestInternal(
 	return resp, err
 }
 
+func (sc *StorageSmartContract) fetchPools(inputBlobbers *StorageNodes, balances chainstate.StateContextI) ([]*blobberWithPool, error) {
+	blobbers := make([]*blobberWithPool, 0, len(inputBlobbers.Nodes))
+	pools := make(chan *blobberWithPool)
+	errs := make(chan error)
```

Comment:
init the channels with buffers, otherwise we'll have gorountines leak if error happens.

Reply:
So there is the same leak in [getAllocationBlobbers](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/storagesc/allocation.go#L549), I'll do a hot fix.